core/src/main/java/com/graphhopper/routing/ch/PrepareContractionHierarchies.java

```diff
@@ -397,97 +367,58 @@ public long getDijkstraCount() {
         return nodeContractor.getDijkstraCount();
     }
 
-    public int getShortcuts() {
+    public long getShortcuts() {
```

Comment:
oh, did you already reached this limit?

Reply:
No, luckily not, but I think its better to be prepared ;)

Reply:
Ah, ok. 

btw: before shortcuts I expect that the edgeIds count will introduce a problem ;( so far we are at 250 million edges and <200 million shortscuts which is ~4x away from the unsigned int limit...

Reply:
Ok. For edge-based CH the number of shortcuts is typically larger than the number of edges, I am still working on reducing the number of shortcuts, but at the moment there are about 3x as many shortcuts than edges for Germany (30 vs 10million). Note that the shortcuts are (so far) unidirected while most of the edges are bidirectional.

Reply:
But you are right that the `getShortcuts()` method is only a simple counter while if this overflows there will be other problems too (edge ids are integers).